src/d/actor/d_a_leaflift.cpp

```diff
@@ -4,98 +4,348 @@
+    fopAcM_setCullSizeFar(this, 1.0f);
+    mStts.Init(0xFF,0xFF,this);
+    mCyl.Set(l_cyl_src);
+    mCyl.SetStts(&mStts);
+    cXyz waterCheckPos = current.pos;
+    waterCheckPos.y += 200.0f;
+    mWaterY = dBgS_ObjGndChk_Wtr_Func(waterCheckPos);
+    if (mWaterY != -1e9f) {
+        cXyz particlePos = current.pos;
+        particlePos.y = mWaterY + 1.0f; 
+        mEmitter3 = dComIfGp_particle_set(0x82AA, &particlePos, &current.angle);
+        if (mEmitter3) {
+            mEmitter3->stopCreateParticle();
+        }
+    }
+    home.pos.y += 15.0f;
+    current.pos.y = home.pos.y;
+    JUTNameTab* jointName = mpModel->getModelData()->getJointName();
+    for (u16 i = 0; i < mpModel->getModelData()->getJointNum(); i++) {
+        if (strcmp("j_happa", jointName->getName(i)) == 0) {
+            mpModel->getModelData()->getJointNodePointer(i)->setCallBack(&nodeCallBack);
+            break;
+        }
+    }
+    mTargetRotation = ZeroQuat;
+    mCurrentRotation = mTargetRotation;
+    if ((s32)daLlift_prm::getPos(this) == 1) {
+        current.pos.y =  home.pos.y + m_height;
+    }
+    else {
+        if (mEmitter3) {
+            mEmitter3->playCreateParticle();
+        }
+    }
+    m43F = 0x1e;
+    mpModel->calc();
+    setMoveBGMtx();
+    dComIfG_Bgsp()->Regist(mpBgW, this);
+    set_mtx();
+    mpBgW->Move();
 }
 
 /* 00000634-00000760       .text _create__9daLlift_cFv */
 s32 daLlift_c::_create() {
-    /* Nonmatching */
+    int res;
+    fopAcM_SetupActor(this, daLlift_c);
+    res = dComIfG_resLoad(&mPhs, m_arcname);
+    if (res == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0xf40)) {
+            return cPhs_ERROR_e;
+        }
+        else {
+            CreateInit();
+        }
+    }
+    return res;
 }
 
 /* 00000918-000009C4       .text nodeCallBack__FP7J3DNodei */
-static BOOL nodeCallBack(J3DNode*, int) {
-    /* Nonmatching */
+static BOOL nodeCallBack(J3DNode* node, int idx) {
+    if (!idx) {
+        J3DJoint* joint = (J3DJoint*)node;
+        s32 jntNo = joint->getJntNo();
+        J3DModel* model = j3dSys.getModel();
+        daLlift_c* i_this = (daLlift_c*)model->getUserArea();
+        if (i_this) {
+            mDoMtx_stack_c::copy(model->getAnmMtx(jntNo));
+            mDoMtx_stack_c::quatM(&i_this->mCurrentRotation);
+            cMtx_copy(mDoMtx_stack_c::get(), J3DSys::mCurrentMtx);
+            model->setAnmMtx(jntNo, mDoMtx_stack_c::get());
+            cMtx_copy(mDoMtx_stack_c::get(), i_this->mBGMtx);
+        }
+    }
+    return TRUE;
 }
 
 /* 000009C4-00000CAC       .text rideCallBack__FP4dBgWP10fopAc_ac_cP10fopAc_ac_c */
-void rideCallBack(dBgW*, fopAc_ac_c*, fopAc_ac_c*) {
-    /* Nonmatching */
+static void rideCallBack(dBgW* param1, fopAc_ac_c* i_this, fopAc_ac_c* i_other) {
+    f32 tiltFactor;
+    f32 offsetMagnitude;
+    daLlift_c* param2 = (daLlift_c*)i_this;
+    daLlift_c* param3 = (daLlift_c*)i_other;
```

Comment:
Incorrect cast, the `fopAc_ac_c* i_other` param to a ride callback can be any type of actor that goes on top of this platform, and it's unlikely that a lilypad platform would land on another lilypad platform.

If you want to cast it, then inside the `if (fopAcM_GetName(param3) == PROC_PLAYER) {` block below you could cast i_other to be of type `daPy_lk_c*` (aka Link's class) because the game explicitly checks that it's Link in that block. (Not really necessary though, since no Link-specific methods are called on this param, so leaving it as `fopAc_ac_c*` is also fine.)